src/node/chainstate.cpp

```diff
@@ -207,7 +207,9 @@ ChainstateLoadResult LoadChainstate(ChainstateManager& chainman, const CacheSize
     } else if (snapshot_completion == SnapshotCompletionResult::SUCCESS) {
         LogPrintf("[snapshot] cleaning up unneeded background chainstate, then reinitializing\n");
         if (!chainman.ValidatedSnapshotCleanup()) {
-            AbortNode("Background chainstate cleanup failed unexpectedly.");
+            const auto error = _("Background chainstate cleanup failed unexpectedly.");
+            AbortNode(error.original);
+            return {ChainstateLoadStatus::INTERRUPTED, error};
```

Comment:
In commit "chainstate: Return on ValidatedSnapshotCleanup fatal failure" (360fe5a47f75514621ef62bc4c906f2112d3308f)

I think this should return FAILURE not INTERRUPTED so caller can know that an error happened, not just an early shutdown request.

But I'd actually suggest dropping this commit from the PR. #27862 should cover this case, and I opened it earlier because I think AbortNode calls in assumeutxo code and in flush code seem different. There should be less of a rationale for ignoring errors in assumeutxo.

Reply:
Yup, I'll drop, make this PR about the flush errors only and continue discussion on your PR.